Content.Server/Traits/TraitSystem.cs

```diff
@@ -45,35 +74,37 @@ private void OnPlayerSpawnComplete(PlayerSpawnCompleteEvent args)
                 continue;
             }
 
-            if (traitPrototype.Whitelist != null && !traitPrototype.Whitelist.IsValid(args.Mob))
+            if (traitPrototype.Whitelist != null && !traitPrototype.Whitelist.IsValid(mob))
             {
                 // You don't get any traits for breaking the whitelist
                 if (traitPrototype.Category == "Negative")
                 {
                     addTraits = false;
                     break;
                 }
-                else continue;
+                else
+                    continue;
             }
 
-            if (traitPrototype.Blacklist != null && traitPrototype.Blacklist.IsValid(args.Mob))
+            if (traitPrototype.Blacklist != null && traitPrototype.Blacklist.IsValid(mob))
             {
                 // You don't get any traits for breaking the blacklist
                 if (traitPrototype.Category == "Negative")
                 {
                     addTraits = false;
                     break;
                 }
-                else continue;
+                else
+                    continue;
             }
 
             traits.Add(traitPrototype);
         }
 
         if (!addTraits)
         {
-            Logger.Warning($"Not adding traits to {args.Player.Name} because they broke the whitelist/blacklist of a negative trait.");
-            return;
+            Logger.Warning($"Not adding traits to entity {mob} because they broke the whitelist/blacklist of a negative trait.");
+            return false;
         }
```

Comment:
This should give some player feedback, even if it's minor. Something pasted in chat or whatever could be good, maybe a message from [server]. Just so they have a way of knowing what happened.